### PR TITLE
Include reactions on issue comments

### LINKED ISSUE #1295: Add reactions to issue comments response payload
Github have added reactions to the issues, comments and pull requests. It would be a really cool feature to have, it's however opt in via header for now.

https://developer.github.com/changes/2016-05-12-reactions-api-preview/
https://developer.github.com/v3/issues/comments/#list-comments-on-an-issue

> An additional reactions object in the issue comment payload is currently available for developers to > preview. During the preview period, the APIs may change without advance notice. Please see the blog post for full details.

> To access the API you must provide a custom media type in the Accept header:
application/vnd.github.squirrel-girl-preview
The reactions key will have the following payload where url can be used to construct the API location for listing and creating reactions.

```
{
  "total_count": 5,
  "+1": 3,
  "-1": 1,
  "laugh": 0,
  "confused": 0,
  "heart": 1,
  "hooray": 0,
  "url": "https://api.github.com/repos/octocat/Hello-World/issues/comments/1/reactions"
}
```

### PR DESCRIPTION
**Pull request by alfhenrik**
- [x] Add Reactions object to IssueCommentsClient
- [x] Add Reactions object to ObservableIssueCommentsClient
- [x] Add tests
- [x] Add integration tests
- [x] Fix builds due to broken tests
- [x] Come to think of it, `IssueCommentReactions` should be renamed, so it can be used on the other relevant payloads.
- [x] Use `AcceptHeaders.ReactionsPreview` instead of local `ReactionsPreview` constant (once #1335 has been merged)

Fixes #1295

### CODE REVIEW

**Comment by alfhenrik** [comment]
RE: renaming `IssueCommentReactions`, perhaps just naming it `Reactions` is better? 

I know #1335 is adding a `Reaction` model, but that is different to this one...but is it going to get confusing?
 
@shiftkey @ryangribble thoughts?

**Comment by ryangribble** [comment]
Yeah naming things is hard (tm) :trollface: 

Perhaps this one in the response payloads could be called `ReactionSummary` since it has counts of each type and the a TotalCount etc.

**Comment by ryangribble** [comment]
Looking good... I guess we'll wait on #1335 for the `AcceptHeaders` value...

**Comment by alfhenrik** [comment]
Yep, or I could add the `AcceptsHeader` in this PR (which would cause a slight merge conflict in #1335), but it looks like #1335 is almost ready for a merge so I have no problem with waiting 😄

**Comment by ryangribble** [comment]
yep I was gonna say you could just add that here too, if you go by the same name as #1335 it wont be much of an issue

**Comment by alfhenrik** [comment]
🎉 look at all those 💚 builds! Clearly @shiftkey's `mono` fix worked 😁

**Comment by ryangribble** [comment]
@alfhenrik #1335 is now merged so you can merge/rebase and update your accepts header usage :+1:

**Comment by alfhenrik** [comment]
👍 will do it tonight

**Comment by alfhenrik** [comment]
PR updated to use new `AcceptHeaders.ReactionsPreview` instead of local variable. Just waiting for 💚 💚 from the running builds

File: Octokit.Tests.Integration/Clients/IssueCommentsClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        var github = Helper.GetAuthenticatedClient();
+        _issueCommentsClient = github.Issue.Comment;
+    }
+
+    [IntegrationTest]
+    public async Task CanDeserializeIssueCommentWhenGettingAllForRepository()
+    {
+        var comments = await _issueCommentsClient.GetAllForRepository("alfhenrik-test", "repo-with-issue-comment-reactions");
+
+        Assert.True(comments.Count > 0);
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/IssueCommentsClientTests.cs]: :lipstick: `Assert.NotEmpty(comments)` ?

Comment (alfhenrik) [inline Octokit.Tests.Integration/Clients/IssueCommentsClientTests.cs]: Yes of course...not sure what I was thinking there 😕

**Comment by shiftkey** [comment]
Just waiting on this to go green, everything else is :gem:

**Comment by shiftkey** [comment]
Fantastic stuff @alfhenrik!

**Comment by alfhenrik** [comment]
Oh, you are too kind sir! 😄

**Pull request merged**